compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/FoldConstantLowering.kt

```diff
@@ -176,6 +174,18 @@ class FoldConstantLowering(private val context: JvmBackendContext) : IrElementTr
                     else -> expression
                 }
             }
+
+            override fun visitStringConcatenation(expression: IrStringConcatenation): IrExpression {
+                expression.transformChildrenVoid(this)
+                return if (expression.arguments.all { it is IrConst<*> && it.value is String }) {
```

Comment:
@pyos Seems it's possible to support some additional cases when begining of IrStringConcatenation is constant concatenation:
expression.arguments.split {!(it is IrConst<*> && it.value is String)

Reply:
@pyos Please also take into account #2244 that also could unmute several tests

Reply:
>Seems it's possible to support some additional cases when begining of IrStringConcatenation is constant concatenation

Maybe fold all constant subsequences then?

>Please also take into account #2244 that also could unmute several tests

I've checked, doesn't look like there is a test for a combination of these changes. Probably because there's no way to produce a constant of type `String?`, so a test would have to do something like `fun foo(x: String?) = x + "a" + "b"` and the non-IR backend does not optimize partially constant concatenations.

Reply:
> Maybe fold all constant subsequences then?
Yes